### ISSUE TITLE
Unexpected log output from Logstash in 1.5RC2

### ISSUE THREAD

**Issue by pickypg** [elastic/logstash#2865]
The relevant Logstash config:

```
# host is filled with an array of valid, statically defined hosts
elasticsearch { 
  action => create 
  flush_size => 1000 
  host => [ ... ] 
  index => "%{type}-%{+YYYY.MM.dd}"
  protocol => "http" 
  workers => 6 
  user => "xxxxx" 
  password => "xxxxx" 
}
```

The unexpected and suspicious log output for _unused_, deprecated elasticsearch settings:

```
{:timestamp=>"2015-03-10T15:02:39.680000-0700", :message=>"You are using a deprecated config setting \"type\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == \"sometype\" { elasticsearch { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"type", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.682000-0700", :message=>"You are using a deprecated config setting \"tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if \"sometag\" in [tags] { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.683000-0700", :message=>"You are using a deprecated config setting \"exclude_tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !(\"sometag\" in [tags]) { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"exclude_tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.683000-0700", :message=>"You are using a deprecated config setting \"max_inflight_requests\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"max_inflight_requests", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.691000-0700", :message=>"You are using a deprecated config setting \"type\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == \"sometype\" { elasticsearch { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"type", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.692000-0700", :message=>"You are using a deprecated config setting \"tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if \"sometag\" in [tags] { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.693000-0700", :message=>"You are using a deprecated config setting \"exclude_tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !(\"sometag\" in [tags]) { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"exclude_tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.694000-0700", :message=>"You are using a deprecated config setting \"max_inflight_requests\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"max_inflight_requests", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.702000-0700", :message=>"You are using a deprecated config setting \"type\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == \"sometype\" { elasticsearch { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"type", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.704000-0700", :message=>"You are using a deprecated config setting \"tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if \"sometag\" in [tags] { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.705000-0700", :message=>"You are using a deprecated config setting \"exclude_tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !(\"sometag\" in [tags]) { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"exclude_tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.706000-0700", :message=>"You are using a deprecated config setting \"max_inflight_requests\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"max_inflight_requests", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.715000-0700", :message=>"You are using a deprecated config setting \"type\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == \"sometype\" { elasticsearch { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"type", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.716000-0700", :message=>"You are using a deprecated config setting \"tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if \"sometag\" in [tags] { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.717000-0700", :message=>"You are using a deprecated config setting \"exclude_tags\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !(\"sometag\" in [tags]) { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"exclude_tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn} 
{:timestamp=>"2015-03-10T15:02:39.718000-0700", :message=>"You are using a deprecated config setting \"max_inflight_requests\" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. If you have any questions about this, please visit the #logstash channel on freenode irc.", :name=>"max_inflight_requests", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn}
```

**Comment by pickypg** [comment]
I made this at the logstast-output-elasticsearch plugin repo where it belongs.

**Comment by jsvd** [comment]
This issue is not plugin specific, it can occur with any of them. Setting workers > 1 triggers this:

`input { generator { count => 1 } } output { stdout { workers => 2} }`:

```
bin/logstash -f config
You are using a deprecated config setting "type" set in stdout. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == "sometype" { stdout { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"type", :plugin=><LogStash::Outputs::Stdout --->, :level=>:warn}
Logstash startup completed
You are using a deprecated config setting "tags" set in stdout. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if "sometag" in [tags] { stdout { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"tags", :plugin=><LogStash::Outputs::Stdout --->, :level=>:warn}
You are using a deprecated config setting "exclude_tags" set in stdout. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !("sometag" in [tags]) { stdout { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"exclude_tags", :plugin=><LogStash::Outputs::Stdout --->, :level=>:warn}
You are using a deprecated config setting "type" set in stdout. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == "sometype" { stdout { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"type", :plugin=><LogStash::Outputs::Stdout --->, :level=>:warn}
You are using a deprecated config setting "tags" set in stdout. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if "sometag" in [tags] { stdout { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"tags", :plugin=><LogStash::Outputs::Stdout --->, :level=>:warn}
You are using a deprecated config setting "exclude_tags" set in stdout. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !("sometag" in [tags]) { stdout { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"exclude_tags", :plugin=><LogStash::Outputs::Stdout --->, :level=>:warn}
Logstash shutdown completed
Exception: Exception
```

**Comment by jsvd** [comment]
closed by 3763ecb
